data_structures/avltree.cpp

```diff
@@ -115,26 +150,42 @@ node *deleteNode(node *root, int key) {
     return root;
 }
 
-/** LevelOrder (Breadth First Search) */
+/**
+ * @brief calls delete on every node
+ * @param root of the tree
+ */
+void deleteAllNodes(const node *const root) {
+    if (root) {
+        deleteAllNodes(root->left);
+        deleteAllNodes(root->right);
+        delete root;
+    }
+}
+
+/**
+ * @brief prints given tree in the LevelOrder
+ * @param root of the tree
+ */
 void levelOrder(node *root) {
     std::queue<node *> q;
     q.push(root);
     while (!q.empty()) {
         root = q.front();
         std::cout << root->data << " ";
         q.pop();
-        if (root->left)
+        if (root->left) {
             q.push(root->left);
-        if (root->right)
+        }
+        if (root->right) {
             q.push(root->right);
+        }
     }
 }
 
-/** Main function */
 int main() {
```

Comment:
```suggestion
/**
 * @brief Main function
 * @returns 0 on exit
 */
int main() {
```

Reply:
I am not sure if the `main` function really has to be documented.

Reply:
It is best to have it documented, IMO.